### ISSUE TITLE
sum/avg/mean ops of tf.keras.backend do not work correctly over batch axis

### ISSUE THREAD

**Issue by joba01** [tensorflow/tensorflow#45656]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.3.0-54-gfcc4b966f1 2.3.1 (docker)
- Python version: 3.6.9
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1.243/7.6.4.38-1
- GPU model and memory: 1080-TI

**Describe the current behavior**
When adding an aggregation function on the batch axis (axis=0), everything is fine, as long as the batchsize is <=32, with batchsize >32 the output shape changes.

Code and test tp reproduce on:
https://stackoverflow.com/questions/64893958/sum-avg-mean-ops-of-tf-keras-backend-do-not-work-correctly-over-batch-axis

**Describe the expected behavior**
output shape is the expected on from model summary and not depending on the batchsize.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
see https://stackoverflow.com/questions/64893958/sum-avg-mean-ops-of-tf-keras-backend-do-not-work-correctly-over-batch-axis

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.2, TF v2.3 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/9cc299da89b22d5aeeea92a97b5cd7b3/45656.ipynb). Thanks!

**Comment by rmothukuru** [comment]
@joba01,
Using **`Relu Activation Layer`** instead of **`keras.backend.max`** gives a reliable `Output` and IMO is recommended, because even the Documentation for [tf.keras.backend.max](https://www.tensorflow.org/api_docs/python/tf/keras/backend/max?hl=FA) is not available in the Tensorflow site, as it is disabled in the [Source Code](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L2272).

Please find [the Gist](https://colab.research.google.com/gist/rmothukuru/bb2e9b3747622c445e1ec8588fd1a6dd/45656.ipynb#scrollTo=u1w8q9D1Bzrq) of working code. Thanks!

**Comment by joba01** [comment]
thanks, but as in the title this is the same for sum/avg/mean. Also Relu is not the same as max. The function is also not deprecated in your linked source.

**Comment by chenmoneygithub** [comment]
Thanks for reporting this issue! 

This error is caused by model.predict(). Basically inside this method your data is split into several epochs by the first axis (the batch axis), and the processed results of each epoch are concatenated together as the final output. This is based on the assumption that all operations in keras.model or keras.layer should not touch the batch axis. For example, suppose we have input data of size (63, 2), when calling predict(), the data is handled as two frames of size (32, 2) and (31, 2), then their outputs are concatenated, which makes the output value of shape (4, ).

One workaround is to use \_\_call\_\_() method instead of predict(). In your example, instead of calling
```
x = tf.random.normal((orders ,parameters))
result = model.predict(x)
print(result.shape)
```
use this
```
x = tf.random.normal((orders ,parameters))
result = model(x)
print(result.shape)
```

**Comment by joba01** [comment]
@chenmoneygithub, thanks a lot for that clarification. I have read that input and output of a model must have the same batchsize/axis-0, but still couldn't explain this behaviour.
